hooks/usePrivyCollect.tsx

```diff
@@ -0,0 +1,58 @@
+import useConnectedWallet from "./useConnectedWallet"
+import { BigNumber } from "ethers"
+import { CHAIN_ID, IS_TESTNET, BASE_MINTER, SEPOLIA_MINTER, MINT_REFERRAL } from "@/lib/consts"
+import { defaultAbiCoder } from "ethers/lib/utils"
+import usePrivySendTransaction from "./usePrivySendTransaction"
+import zora1155Abi from "@/lib/abi/abi-ERC1155Drop.json"
+import usePreparePrivyWallet from "./usePreparePrivyWallet"
+import { toast } from "react-toastify"
+import useZoraFixedPriceSaleStrategy from "./useZoraFixedPriceSaleStrategy"
+import { ZORA_FEE } from "@/lib/consts"
+import handleTxError from "@/lib/handleTxError"
+import { useState } from "react"
+
+const usePrivyCollect = () => {
+  const { sendTransaction } = usePrivySendTransaction()
+  const { prepare } = usePreparePrivyWallet()
+  const { connectedWallet } = useConnectedWallet()
+  const [loading, setLoading] = useState(false)
+
+  const { sale } = useZoraFixedPriceSaleStrategy({
+    saleConfig: IS_TESTNET ? SEPOLIA_MINTER : BASE_MINTER,
+    chainId: CHAIN_ID,
+  })
+
+  const collectAll = async () => {
+    setLoading(true)
+    try {
+      if (!prepare()) return
+      if (!connectedWallet) return
```

Comment:
I think you should more this check for `!connectedWallet` into the prepare hook right?

Or, do you think it's better here? 🤔